.gitignore

```diff
@@ -4,3 +4,4 @@ Cargo.lock
 /examples/viam.json
 /canary/.venv/*
 .vscode
+components_esp32.lock
```

Comment:
Thanks for adding this. Do we know what this file does?

Reply:
It looks like it's an auto-generated list of components for a project?